corehq/apps/case_search/filter_dsl.py

```diff
@@ -149,6 +149,10 @@ def build_filter_from_xpath(xpath, *, domain=None, context=None):
             raise CaseFilterError(error_message.format(bad_part, ", ".join(ALL_OPERATORS)), bad_part)
         raise CaseFilterError(_("Malformed search query"), None)
     except RuntimeError as e:
+        # eulxml doesn't appear to clean up after this type of failure
+        # properly, so throw in an extra 'parse' to reset it
+        parse_xpath("thisisdumb")
```

Comment:
Really the fix here is for us to get off of `eulxml`, which we've wanted to do a couple times in the past as well.  Just made a backlog item for that:
https://dimagi.atlassian.net/browse/USH-5070